lib/Transformation.ts

```diff
@@ -190,20 +189,19 @@ function wrapAsyncFunction(f: AsyncExtensionFunction, name: string): AsyncExtens
   });
 }
 // TODO: Support passing functions to override default behaviour;
-function transformNamed<SyncType extends  boolean>(expr: Alg.NamedExpression,
-                                                   extensionCallback: ExtensionCallbackType<SyncType>, sync: SyncType)
+function transformNamed(expr: Alg.NamedExpression, creatorConfig: FunctionCreatorConfig)
   : E.NamedExpression | E.AsyncExtensionExpression | E.SyncExtensionExpression  {
   const funcName = expr.name.value;
-  const args = expr.args.map((a) => transformAlgebra(a, extensionCallback, sync));
+  const args = expr.args.map((a) => transformAlgebra(a, creatorConfig));
   if (!C.NamedOperators.contains(funcName as C.NamedOperator)) {
```

Comment:
Now we have this cleaned up a bit, we change the order of the paths and add some documentation.

Notice how the base case (a known named function) is now relegated to the end, with a lot of branching in between (even with negated expressions). I suggest moving this to the beginning, i.e. if it exists, great, return early. Then we remove a level of nesting for function extensions, which can come afterwards. Also add a little documentation to these branches.